### PR TITLE
added aix_nimclient module

### PR DESCRIPTION
**Pull request by molekuul**
##### SUMMARY

This new module is ment for AIX. It installs software at a nimclient using nimclient command and removes software using local commands. This module also updates ALL to latest version, and ia able to issues nimclient action on the nimclient like reset or deallocate

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

aix_nimclient

##### ADDITIONAL INFORMATION

Examples:
```yaml
- name: install latest version of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06
  aix_nimclient:
    name:
      - OpenGL.OpenGL_X.rte.soft
    lpp_source: lppsource_aix6109-06

- name: install latest version of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06 and apply only
  aix_nimclient:
    name:
      - OpenGL.OpenGL_X.rte.soft
    lpp_source: lppsource_aix6109-06
    commit: False

- name: install version 6.1.9.30 of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06
  aix_nimclient:
    name:
      - OpenGL.OpenGL_X.rte.soft 6.1.9.30
    lpp_source: lppsource_aix6109-06

- name: remove fileset OpenGL.OpenGL_X.rte.soft
  aix_nimclient:
    name:
      - OpenGL.OpenGL_X.rte.soft
    state: absent

- name: commit fileset OpenGL.OpenGL_X.rte.soft
  aix_nimclient:
    name:
      - OpenGL.OpenGL_X.rte.soft
    state: commit

# install all filesets to latest level f.i. install a TL or ML
- name: update all filesets to latest level from lpp_source lppsource_aix6109-06
  aix_nimclient:
    name:
      - update_all
    lpp_source: lppsource_aix6109-06

- name: allocate spot and lpp_source to the nimclient
  aix_nimclient:
    lpp_source: lppsource_aix6109-06
    spot: spot_aix6109-06
    state: allocate

- name: deallocate all resources from the nimclient
  aix_nimclient:
    state: deallocate

- name: reset the nimclient
  aix_nimclient:
    state: reset


```

### CODE REVIEW

File: lib/ansible/modules/system/aix_nimclient.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+author: "Joris Weijters (@molekuul)"
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
author:
- Joris Weijters (@molekuul)
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
DOCUMENTATION = r'''
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+---
+author: "Joris Weijters (@molekuul)"
+module: aix_nimclient
+short_description: installs software using the nimclient, and removes software
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
short_description: Manage software using the nimclient
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+description:
+    - installs software at the nimclient using nimclient command and removes software using local commands also updates ALL to latest version
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    - The module installs software using nimclient command, removes software using local commands, and updates all software to latest available version.
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 5]
```diff
@@ -0,0 +1,545 @@
...
+version_added: "2.8"
+options:
+  name:
+    description:
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py:25]:
```suggestion
    description:
    - Name of the fileset to install.
    - Use C(update_all) to update all filesets to latest level.
    type: list
```
Other modules use `*` or `all`, so maybe that's a better convention to use ?

File: lib/ansible/modules/system/aix_nimclient.py [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+    - Name of the fileset to install or update_all to update all filesets to latest level
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+    aliases:
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    aliases: [ filesets ]
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+      - filesets
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 9]
```diff
@@ -0,0 +1,545 @@
...
+    type: list
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py:28]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 10]
```diff
@@ -0,0 +1,545 @@
...
+  state:
+    description:
+    - State of the fileset or nimclient to be
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    - The state of the fileset or nimclient to be.
    type: str
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 11]
```diff
@@ -0,0 +1,545 @@
...
+    choices: [
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    choices: [ absent, allocate, commit, deallocate, installed, present, removed, reset ]
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 12]
```diff
@@ -0,0 +1,545 @@
...
+              commit,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 13]
```diff
@@ -0,0 +1,545 @@
...
+              present,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 14]
```diff
@@ -0,0 +1,545 @@
...
+              absent,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 15]
```diff
@@ -0,0 +1,545 @@
...
+              installed,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 16]
```diff
@@ -0,0 +1,545 @@
...
+              removed,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 17]
```diff
@@ -0,0 +1,545 @@
...
+              allocate,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 18]
```diff
@@ -0,0 +1,545 @@
...
+              deallocate,
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 19]
```diff
@@ -0,0 +1,545 @@
...
+              reset
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 20]
```diff
@@ -0,0 +1,545 @@
...
+              ]
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 21]
```diff
@@ -0,0 +1,545 @@
...
+    default : present
+    type: str
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py:33]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 22]
```diff
@@ -0,0 +1,545 @@
...
+  lpp_source:
+    description:
+    - Name of the lpp_source at the nimserver to be used for installation
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    - Name of the C(lpp_source) at the C(nimserver) to be used for installation.
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 23]
```diff
@@ -0,0 +1,545 @@
...
+    type: str
+  spot:
+    description:
+    - Name of the Spot at the nimserver
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    - Name of the Spot at the C(nimserver).
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 24]
```diff
@@ -0,0 +1,545 @@
...
+    type: str
+  commit:
+    description:
+    - boolean for committing the fileset during install or not
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    - Whether to commit the fileset during install or not.
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 25]
```diff
@@ -0,0 +1,545 @@
...
+    type: bool
+    default: True
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    default: yes
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 26]
```diff
@@ -0,0 +1,545 @@
...
+  installp_flags:
+    description:
+    - installp flags
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    - Additional C(installp) flags to use.
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 27]
```diff
@@ -0,0 +1,545 @@
...
+    type: str
+    default: 'acgwXY'
+
+notes:
+  - The changes are persistent across reboots.
+  - You need root rights to install or remove software
+  - tested on AIX 6.1 and 7.1.
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
  - Tested on AIX 6.1 and 7.1.
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 28]
```diff
@@ -0,0 +1,545 @@
...
+requirements: [ 're' ]
+'''
+
+EXAMPLES = '''
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
EXAMPLES = r'''
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 29]
```diff
@@ -0,0 +1,545 @@
...
+- name: install latest version of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Install latest version of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 30]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    name:
+      - OpenGL.OpenGL_X.rte.soft
+    lpp_source: lppsource_aix6109-06
+
+- name: install latest version of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06 and apply only
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Install latest version of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06 and apply only
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 31]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    name:
+      - OpenGL.OpenGL_X.rte.soft
+    lpp_source: lppsource_aix6109-06
+    commit: False
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    commit: no
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 32]
```diff
@@ -0,0 +1,545 @@
...
+
+- name: install version 6.1.9.30 of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Install version 6.1.9.30 of OpenGL.OpenGL_X.rte.soft from lpp_source lppsource_aix6109-06
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 33]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    name:
+      - OpenGL.OpenGL_X.rte.soft 6.1.9.30
+    lpp_source: lppsource_aix6109-06
+
+- name: remove fileset OpenGL.OpenGL_X.rte.soft
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Remove fileset OpenGL.OpenGL_X.rte.soft
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 34]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    name:
+      - OpenGL.OpenGL_X.rte.soft
+    state: absent
+
+- name: commit fileset OpenGL.OpenGL_X.rte.soft
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Commit fileset OpenGL.OpenGL_X.rte.soft
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 35]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    name:
+      - OpenGL.OpenGL_X.rte.soft
+    state: commit
+
+# install all filesets to latest level f.i. install a TL or ML
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
# Install all filesets to latest level f.i. install a TL or ML
```
Not sure what all this means.

File: lib/ansible/modules/system/aix_nimclient.py [hunk 36]
```diff
@@ -0,0 +1,545 @@
...
+- name: update all filesets to latest level from lpp_source lppsource_aix6109-06
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Update all filesets to latest level from lpp_source lppsource_aix6109-06
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 37]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    name:
+      - update_all
+    lpp_source: lppsource_aix6109-06
+
+- name: allocate spot and lpp_source to the nimclient
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Allocate spot and lpp_source to the nimclient
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 38]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    lpp_source: lppsource_aix6109-06
+    spot: spot_aix6109-06
+    state: allocate
+
+- name: deallocate all resources from the nimclient
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Deallocate all resources from the nimclient
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 39]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    state: deallocate
+
+- name: reset the nimclient
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
- name: Reset the nimclient
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 40]
```diff
@@ -0,0 +1,545 @@
...
+  aix_nimclient:
+    state: reset
+'''
+
+RETURN = '''
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
RETURN = r'''
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 41]
```diff
@@ -0,0 +1,545 @@
...
+                msg=msg, err=err, rc=rc)
+    return result
+
+
+def main():
+    # initialize
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='list', aliases=['filesets']),
+            state=dict(choices=[
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
            state=dict(type='list', default='present', choices=['absent', 'allocate', 'commit', 'deallocate', 'installed', 'present', 'removed', 'reset']),
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 42]
```diff
@@ -0,0 +1,545 @@
...
+                'present',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 43]
```diff
@@ -0,0 +1,545 @@
...
+                'commit',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 44]
```diff
@@ -0,0 +1,545 @@
...
+                'absent',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 45]
```diff
@@ -0,0 +1,545 @@
...
+                'installed',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 46]
```diff
@@ -0,0 +1,545 @@
...
+                'removed',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 47]
```diff
@@ -0,0 +1,545 @@
...
+                'allocate',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 48]
```diff
@@ -0,0 +1,545 @@
...
+                'deallocate',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 49]
```diff
@@ -0,0 +1,545 @@
...
+                'reset',
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 50]
```diff
@@ -0,0 +1,545 @@
...
+            ], default='present'),
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 51]
```diff
@@ -0,0 +1,545 @@
...
+            lpp_source=dict(type='str'),
+            spot=dict(type='str'),
+            commit=dict(type='bool', default='True'),
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
            commit=dict(type='bool', default=True),
```

File: lib/ansible/modules/system/aix_nimclient.py [hunk 52]
```diff
@@ -0,0 +1,545 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Joris Weijters <joris.weijters@gmail.com>
```
Comment (dagwieers) [inline lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
# Copyright: (c) 2017, Joris Weijters (@molekuul) <joris.weijters@gmail.com>
```

**Review by dagwieers — DISMISSED** [review]
Trying to make multiline edits with the Github suggestion mechanism was rather painful :-(

These are mostly cosmetic changes at this time. Will do second review of the code after this has been accepted.

File: lib/ansible/modules/system/aix_nimclient.py (hunk 4 above)
Comment (gforster) [reply lib/ansible/modules/system/aix_nimclient.py]:
```suggestion
    -The module installs software using nimclient command, removes software using local commands, and updates all software to latest available version.
```

Oxford comma for clarity

Comment (dagwieers) [reply lib/ansible/modules/system/aix_nimclient.py]: @gforster Beware you broke YAML there ! But I put your suggestion into mine.

Comment (gforster) [reply lib/ansible/modules/system/aix_nimclient.py]: Ah! thanks for the catch. I need to be more careful with this editor.

**Comment by molekuul** [comment]
Well that was painfull..... this tool isn't just not made for these kind of small adjustments.
I don't know if all are correct now. I will run tests at my own system, shippable will take some time to process all 50+ commits :-)

kind regards,

Joris Weijters

**Comment by dagwieers** [comment]
bot_skip

**Comment by molekuul** [comment]
@dagwieers 

I did look at all proposed changes, and included then into the module however the status is still Changes requested. What else has to be done to get this module in Ansible?

Kind regards,

Joris Weijters

**Comment by dagwieers** [comment]
@molekuul That's because I haven't approved it yet. AFAIK you can dismiss my review, i.e. because it is outdated.

Now, before we continue with this specific module, I want to make absolutely sure there is no conflict with the modules from https://github.com/aixoss/ansible-playbooks/tree/master/library and we follow similar naming-conventions for the user-interface  (module-naming, parameters). I hope that at some point in the future we can merge their modules/functionality into devel branch.

I'd like more people (with an AIX background) to review these modules (beyond the docs/cosmetics that I did) so we have the best possible implementation. And once that is done we can ask people to test it on their infrastructure and add shipits.

If we get shipits early, every change to the branch will invalidate them. So testing should be done on the final codebase.

**Comment by dagwieers** [comment]
There's also a functional overlap with the **installp** module, next to **aix_nim**. So we need to clarify what our ideal offering looks like.

cc @kairoaraujo @wtcross @pbfinley1911

**Comment by molekuul** [comment]
NIM stands for Network Install Manager. It is meant to install AIX by the network.
At a NIM server you define the LPP-source. This is a repository of filesets containing LPP and RPM packages The LPP packages are the traditional AIX packages, the RPM packages are the Linux packages ported to AIX, 

From this LPP-source you install AIX. Multiple LPP_sources can exist.
Installing a AIX server starts with configuring the nimclient at the NIM-Server, and install it. 
Compared to Linux you could say: The NIM-Server is like a Satellite server, and the nimclient is the Linux instance where the nimclient command is like the yum command.

How to patch a AIX server which is a nimclient:
The system manager of the NIM-Server updates the software repositories with the patches. 
Now there are 2 ways to install these patches at the nimclient: 
- push from the NIM-Server
- pull from the nimclient

pushing the patches, means a central management of the clients, and so access to the NIM-Server.
pulling means the owner of the nimclient can patch "his AIX server" without access to the NIM-Server.

In traditional aix system managemt, there is a central team managing the IBM Power hardware, VIO-Server and NIM-Server and nimclients.  
In the cloud, the nimclients are owned by the consumer, So pulling patches is the prefered way.
This module is for installing AIX software by pulling 

There is a little overlap with the installp module: The "nimclient" command can not remove or commit software. However to have this functionality, in the nimclient module this is done by the installp command. 
The installp command is a command to install, update, commit and remove local accessible software packages , if compared to Linux, see it as the rpm command ( however only local packages, no remote http support)

The aixoss packages are written for defining and managing the NIM-Server, and pushing patches to VIO-Servers. A VIO-Server is a special type of server, Virtual I/O Server. This server is used for connecting virtual devices to physical devices. If compared to VMWare, it's like the ESX layer, but it's not the same. 


Kind regards,

Joris Weijters

**Review by dagwieers — APPROVED** [review]
LGTM, although this module needs to be reviewed and tested by AIX users.
*(We could also benefit from having integration tests)*

**Review by flynn1973 — APPROVED** [review]

**Comment by flynn1973** [comment]
ready_for_review

**Comment by flynn1973** [comment]
folks, please review to get it merged before collections makes things even more complicated.

**Comment by Akasurde** [comment]
Hi @molekuul, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**